CRM/Financial/BAO/FinancialTypeAccount.php

```diff
@@ -154,17 +154,19 @@ public static function del($financialTypeAccountId, $accountId = NULL) {
    * @return null|int
    */
   public static function getInstrumentFinancialAccount($paymentInstrumentValue) {
-    $paymentInstrument = civicrm_api3('OptionValue', 'getsingle', array(
-      'return' => array("id"),
-      'value' => $paymentInstrumentValue,
-      'option_group_id' => "payment_instrument",
-    ));
-    $financialAccountId = CRM_Contribute_PseudoConstant::getRelationalFinancialAccount(
-      $paymentInstrument['id'],
-      NULL,
-      'civicrm_option_value'
-    );
-    return $financialAccountId;
+    if (!isset(\Civi::$statics[__CLASS__]['instrument_financial_accounts'][$paymentInstrumentValue])) {
+      $paymentInstrumentID = civicrm_api3('OptionValue', 'getvalue', array(
+        'return' => 'id',
+        'value' => $paymentInstrumentValue,
+        'option_group_id' => "payment_instrument",
+      ));
+      \Civi::$statics[__CLASS__]['instrument_financial_accounts'][$paymentInstrumentValue] = civicrm_api3('EntityFinancialAccount', 'getvalue', [
+        'return' => 'financial_account_id',
+        'entity_table' => 'civicrm_option_value',
+        'entity_id' => $paymentInstrumentID,
+      ]);
```

Comment:
Option Value api doesn't create financial account relationship with payment instrument. Won't getvalue cause exception error for such cases i.e payment instrument with missing financial account relation(though its a different issue and need to handle at api level but still we should avoid fatal errors on screen for those who doesn't bother financial accounting entries). Thoughts?

Reply:
@pradpnayak so you are saying it is valid for this function to return NULL / FALSE?

Reply:
@pradpnayak OK - looks like NULL could be returned before - have updated to re-instate that